### ISSUE TITLE
Get the electronics and software up and running on the monster laser cutter

### ISSUE THREAD

**Issue by goatchurchprime** [DoESLiverpool/somebody-should#607]
Having obtained the extra large laser cutter that hasn't got any software from the Wirral #586 
we need to know what kind it is in order to decide what we're going to do.  

The reason the software is missing is that someone at the school first disposed of all the computers -- including the only one with the software on it.  

We need to substantiate the claim that the replacement software is only available for £6k.  

Once that is done the options range from reverse engineering the communication protocol with the controller from the computer, to completely replacing the controller with something like a beaglebone/machinekit combo or another laser cutting board.

**Comment by amcewen** [comment]
When we were first told about it, we were told it's one of these http://www.cct-uk.com/fb1800.htm

**Comment by goatchurchprime** [comment]
A lot of the software looks like it is for cutting fabric using cameras to spot the wrinkles.  

I think what we'd need is Cutting Composer, which has a lot of features, when we are only after the basic controller part. 

The brochure is vague about the software, but seems to say this FB series is capable of handling chlorine products such as vinyl, as this one is focused on fabrics.  
    http://www.cct-uk.com/pdf/CadCamlaserbrochure2012.pdf

It's related to this software, which does fancy embroidery:
    http://aps-ethos.com/laser_cutting_software.php

This is someone else's guide for using the software and laser cutter:
    https://www.uclan.ac.uk/about_us/facilities/assets/Laser_Cutter_Guide.pdf

We need to get some photos internally of this machine and its motors to see what components make up the controller.

**Comment by KarlDunkerley** [comment]
Hi. I'm new to DoesLiverpool but I do know that the FabLab at Beamont College in Warrington uses an APS-Ethos machine. The manager is James Ingman and the main laser techie is Enrique Buenas. The software can be temperamental especially when importing images and text.

**Comment by goatchurchprime** [comment]
Some photos of the dismantling and move to first floor are here: https://github.com/DoESLiverpool/somewhere-safe/tree/master/HugeLaserCutterMove

**Comment by Thingomy** [comment]
I'm strongly in favour of striping it down, putting a decent DIY friendly
controller in there and then (if sucsessfull) expanding that effort to the
other laser cutters to get rid of the god aweful software we need to use
(before going on to take over the world...) having 3 laser cutters means
that we can afford to have one down for major overhaul processes and still
be able to tolerate routine breakdowns in the others.

I also willing to be involved, though I'm not sure I'd be the right person
to lead this.

Next question: are there any open source options that work?

JR Peterson

**Comment by goatchurchprime** [comment]
There are lots of alternatives around.  Title doesn't say we buy it.  But we should know about it and what it does.  Are we missing just the computer that generates the plot code (in ascii) or are we missing the whole controller system, steppers and everything?

An important part of the phone call will be when they name the price, we have to ask what the work-arounds and alternatives are when we can't afford it.  They could say: Well, it generates G-code which only experts like us know how to generate, or they could say: Well, it's locked down with proprietary closed source encrypted stuff so you can't work around it.  These are quite different answers, and I don't know which they would give.

**Comment by goatchurchprime** [comment]
Someone left the cover off everything and we have a thick layer of dust over everything, electronics, belts, gears, etc.  It's going to need a serious stripdown and cleaning now.

Meanwhile, I have looked at the basic issue with the electronics.  Externally it's run from an RS232 cable from the outer computer.  This probably downloads a whole G-code or HPGL file to the internal computer (now full of dust), which has a couple of boards, the top one probably just to control the display.

The XY motions are run by some very small servo motors.  The X is a pair of servos with two belts onto the same drive wheel.  This is pretty weird, but probably a hack when one of them was underpowered.  There is a "Computer Optical Products Inc" encoder on one of these motors.  

Our options remaining are to:

 (a) reassemble this computer and attempt to reverse engineer the code that is downloaded to it from the controller computer (the one that is missing).  We could get help with this if we can find a similar machine (eg in Warrington) and have the capability to sniff the data that's going 2-way through their RS232 cable.  I guess this would be like a splitter wire we could put in-line with their cable (using the sockets) and a computer that takes the baudrate and streams out the bytes into a file for inspection.  

 (b) strip out the computer and replace it with machinekit on a beaglebone using one of the boards used on the machine tool and some spare servo motor drivers we have kicking around.  It would be best to configure as much as possible through the machinekit config files (like homing spots, end switches, laser-on or off (eg using spindle speed or z-height as proxy for intensity, who knows), etc).  

 (c) find some other laser controller appropriate system like machinekit (or adapted from machinekit) which has the features of varying the laser intensity according to the feed speed, so it does corners properly.

**Comment by KarlDunkerley** [comment]
Just been told by Simeon Diaz at CCT-UK that; "Machine SN 051004FB676: is FB Laser - FB1550 (Sheet Only) with Universal 50AC laser". 

Here's a link to the product page on their website;

http://www.cct-uk.com/fb1500.php

**Comment by magman2112** [comment]
When we took this machine apart to move it upstairs, I remember quite a few photographs being taken of the machine, to help with it’s reassembly. Are these collected or hosted anywhere at present? If not, can they be?

I can likely be available on Saturday this week if some folks want to have a go at putting this machine back together again.

**Comment by skos-ninja** [comment]
When it was moved @jackie1050 took all the photos however I'm not sure where they are now

**Comment by KarlDunkerley** [comment]
We need to get the latest addition to our laser-powered fleet up and running. Here's what I have discovered so far thanks to simeon.diaz@cct-uk.com who supplied the machine originally;

"The price for the software will be £1565 +VAT + delivery. The price includes the software Aps-Ethos V17 and a USB dongle. The software will allow you to create the designs and send them to the cutter.

It includes 1 year software support if you find any bugs we will try to resolved them.
Our software is backwards compatible but the warranty will be limited considering the age of your machine (2004). In the unforeseen event that Aps-Ethos V17 can’t be fixed to work with your machine we will refund you the cost of the software, after receiving the new dongle."

![38543181-c3de64a0-3c9b-11e8-8851-4611d28a3ac5](https://user-images.githubusercontent.com/34492815/38614110-ede54302-3d83-11e8-896a-73a5cc33a599.jpg)

**Comment by KarlDunkerley** [comment]
Comment by Thingommy;

I like the name.

I regard this price as utterly insane -- especially given the ethos (pun fully intended) of maker spaces. Do we have any experaince of open source (or even other comercial) options?

I would sugest that priorities for selection would need to include:

compatible with hardware
user friendly and easy to pick up fast given how many users will be involved
fully featured (by some definition)
reliable (not that the old solution is, given the inkscape compatibiltiy issues to name one)
extensable (ie compatible with Sophia and Gerald, should we want to go down that road later)
easilly implemented (at the bottom of the list for a reason, I don't expect that to be very liklely)
A quick search identifies:

LaserGRBL -- Usability wise this isn't great as it reduces us down ot the same workflow that a 3D printer, needs with a separate "slicer" tool. -- I think

Laserweb -- this looks the bomb -- there are some vids on youtube -- uses grbl (or others) on the firmware side, which is fine, but that would mean retrofitting steppers in place of the servoes -- there is liklely a gcode to servo motors firmware option in existance?? -- others around here have more experiance with this than me, so please chip in.

**Comment by KarlDunkerley** [comment]
I've heard on various forums that LightBurn is very good for usability and features eg multiple passes. Also heard it is compatible with quite a lot of kit. There is a trial version and the full version is around £80. All the reviews suggest it is money worth spending. Would be worth a try with the trial version.

**Comment by KarlDunkerley** [comment]
Comment from GoatChurchPrime;

https://lightburnsoftware.com/ supports "most GCode and Ruida based controllers. GCode controllers, including Grbl, Smoothie, and Grbl-LPC, and Ruida controllers, including the RDC6442G, RDC63344, RDLC-320A, and the R5-DSP are among those supported."

https://github.com/LightBurnSoftware/Documentation/blob/master/DeviceWizard.md

What kind of controllers do we have on Sophia and Gerald? They're a pretty common sort, and if they're not in the list, it's very unlikely that this one will be. For obvious reasons, their controller list includes the open source ones (grbl and beaglebone based ones).

One problem is we let plaster dust get into every part of the machinery, it's disgusting. I found the whole thing uncovered one day and thickly layered. Possibly all the belts need taking off and washing. The metal machine tool also got quite a coating as well.

**Comment by amcewen** [comment]
Keith from the Leather Satchel Company happened to call in yesterday, and it turns out he's got a smaller one of these machines at his place.

He's also got a spare dongle for the software, so we could borrow that - at least to see how it interfaces with the machine and/or to find out more about how it works.

**Comment by magman2112** [comment]
I've finally got around to documenting our many discussions regarding software and related controllers for Morris, so here is the current consensus of these discussions.

The original software for Morris is just not economic for DoES, at £1565 plus VAT plus shipping. This then points us in the direction of replacing the controller with a cheaper equivalent and finding matching software.

I found an interesting analysis of Laser Controllers at this site;
https://lasergods.com/controller-dsp-identification/
Ironically, @KarlDunkerley sent me the same link a few days after I found it myself.

I like the idea of the Lightburn software mentioned by Karl above, this has a matching controller from Cohesion3d which would appear to be a good fit for our requirements, at a reasonable cost. https://cohesion3d.com/

This controller is designed as a direct replacement for K40 laser controllers, similar to newer version of laser cutters like Sophia. So even should we find that this is not a good fit for Morris, we have a fallback position of being able to upgrade Sophia. 

Given that Morris is currently using Servo Motors to drive the axes, we would have to use an external motor driver with this controller. Fortunately, @goatchurchprime has a couple of spare controlllers that appear to be capable of supporting the existing motors, after some testing that @goatchurchprime and @Sean-anotherone did last maker night, these controllers can move the motors, though some tuning will be required to make them work as required.

Working on the premise that this is a suitable controller, I came up with the following shopping list of parts for replacing the controller on Morris:-

Cohesion 3D Laserboard controller - $199.99
Graphic LCD Control Panel - $30
GLCD Adapter for Cohesion3D Mini/ LaserBoard - $5
Cohesion3D LaserBoard External Stepper Driver Cable [4 Pack] - $15
Ethernet Expansion Board - $10
Lightburn 4.0 Software - $40

This comes to a total of $304.

Whilst I was exploring the shop at Cohesion, I also found a couple of other items that I think we should pick up at the same time. There now appears to be a camera option for Lightburn that would be interesting to experiment with and they also have replacement carriage wheels that would appear to be the right ones for Sophia.

Official LightBurn Camera - $80
X Carriage Wheel Set for K40 Laser - 2 x $25 ($50)

With these added, the complete order from Cohesion would be $429.99, plus shipping at $42.03 (the cheapest option) for an overall total of $472.02.

Assuming a currency exchange rate of $1.30/£ (this was the rate I got on my credit card last week, though may fluctuate) gives us a total of £363.23.

We are also likely to have to pay VAT and possibly import duty on receipt of the parcel (Plus a parcelforce handling charge), as follows:

VAT at 20% on $429.99 (£330.76) = £66.15
Import duty at 2.7% = £8.93
Post office handling charge = £12.00

This gives us a grand total of £450.31

I'm going to ask for pledges on the DoES Google Group to see if we can get enough interested parties to fund this project and I will report back here on the pledge status.

**Comment by magman2112** [comment]
A comment to capture the pledges so far:

Mike Gorman - £50
Dynamic Devices (Alex Lennon) - £50

Running total - £100

**Comment by Sean-anotherone** [comment]
Put  me down for £50 too.

    On Sunday, 31 March 2019, 23:59:32 BST, Mike Gorman <notifications@github.com> wrote:  
 
 
A comment to capture the pledges so far:

Mike Gorman - £50
Dynamic Devices (Alex Lennon) - £50

Running total - £100

**Comment by magman2112** [comment]
A slight snag has come to my attention in that my VAT and import duty calculations may be off slightly. I had assumed that postal charges are not included in these calculations, but from further research today it appears that they should be included.

My new calculations are as follows :-

VAT at 20% on $472.02 (£363.23) = £72.65
Import duty at 2.7% = £9.81
Post office handling charge = £12.00

This gives a difference of an additional £7.38 we will likely have to pay, for a grand total of £457.69.

**Comment by DoESsean** [comment]
@Sean-anotherone and @Robotorium I don't have email addresses for you, so haven't been able to include you on the email I've just sent.

@magman2112 has asked me to get the ball rolling on pledge collections, so if you can get your pledge to us in whatever method is best for you, we can get started :)

**Comment by DoESsean** [comment]
Thanks thus far to @amcewen, @magman2112, @KarlDunkerley, @ajlennon and Martin Dunschen for their donations.

**Comment by Sean-anotherone** [comment]
Sorted mine too now, found the account details on my original membership email.
Sean  (2072)


    On Tuesday, 7 May 2019, 12:23:35 BST, Seán Gleeson <notifications@github.com> wrote:  
 
 
Thanks thus far to @amcewen, @magman2112, @KarlDunkerley, @ajlennon and Martin Dunschen for their donations.

**Comment by paulgeering** [comment]
I missed the original request for pledges, so went straight to the donation phase.
£20 from me (Bank reference PGEERING-MORRISFUN)

**Comment by magman2112** [comment]
With most of the funds now being safely in the bank @DoESsean ordered all the items on the BOM this afternoon. We plumped for a more expensive postal option, so this should be with us in 6-10 days.

**Comment by Sean-anotherone** [comment]
Has this lot arrived yet or is it caught up somewhere - I seem to recall hearing about 5-10 days post time around 3+ weeks ago... ?

**Comment by DoESsean** [comment]
According to the tracking information online, this got from New Jersey to Heathrow in three days, and has been in transit since. It could have been walked here in that time.

I can't see any way to find out what is going on (as it says that customer service representatives will only have access to the same tracking info as we do). Any ideas?

[https://tools.usps.com/go/TrackConfirmAction_input?qtc_tLabels1=CJ493607998US](url)

**Comment by DoESsean** [comment]
As an update to this, I've emailed Cohesion3d to see if they can help. I can only assume that the package has been held at customs or something, but, as no-one has contacted us, I have no way of knowing if that is definitely the case, or how we go about sorting it out.

I'll keep the thread informed.

**Comment by DoESsean** [comment]
Another (largely worthless) update.

I've once again emailed Cohesion3d, this time using the contact form on their website rather than just a standard email. I have no idea whether that will make a difference.

I've also emailed USPS, although I'm not sure if I've contacted the right area - the only way to get in touch through their website is via a contact form that requires US zip codes and State for the destination address, which obviously we don't have.

Again, I'll let you know what comes of it.

On the plus side (although it doesn't help us with getting Morris up and running) we paid through PayPal, so should be able to make a claim against the money which would prevent us losing out financially from all this.

**Comment by KarlDunkerley** [comment]
Don't know if this will help but when I ordered mine, it was given by USPS to ParcelForce who held it in Bamber Bridge depot for collection and customs duties.
I did get a notification and went to pick it up in person.
Am wondering if there is any way to contact PF and see if they have it at a depot for Liverpool. 
Just a thought....
    On Thursday, 20 June 2019, 12:49:20 BST, Seán Gleeson <notifications@github.com> wrote:  
 
 
Another (largely worthless) update.

I've once again emailed Cohesion3d, this time using the contact form on their website rather than just a standard email. I have no idea whether that will make a difference.

I've also emailed USPS, although I'm not sure if I've contacted the right area - the only way to get in touch through their website is via a contact form that requires US zip codes and State for the destination address, which obviously we don't have.

Again, I'll let you know what comes of it.

On the plus side (although it doesn't help us with getting Morris up and running) we paid through PayPal, so should be able to make a claim against the money which would prevent us losing out financially from all this.

**Comment by DoESsean** [comment]
That might actually be really useful @KarlDunkerley 

I've just looked on Parcelforce's tracking, and they recognise the tracking number. Now, they're saying it's not yet arrived in the UK, so that could be an issue, but I can give them a call.

**Comment by DoESsean** [comment]
An update:

Cohesion3d have not responded to any emails, either directly or through the contact form on their site.

USPS replied asking me to use the form on their site. I told them I was unable to due to not having a State or ZIP, and asking whether it could be handled via email. They have not responded.

I spoke to Parcelforce, the lady on the phone can see exactly what I can - that it's apparently been delivered to Heathrow, but has never been scanned in by Parcelforce, so they haven't received it, meaning there's basically nothing they can do.

I'm going to raise a PayPal complaint to get the money refunded.

Does anyone know of anywhere else we can get the stuff we need?

**Comment by DoESsean** [comment]
Dispute raised. I'll escalate it to a refund if Cohesion haven't come back within a week.

**Comment by DoESsean** [comment]
The PayPal dispute has gotten a quick response.

The owner of Cohesion3d said:

Hi, 

Really sorry about this. I hadn't seen any previous messages from you guys. I did some investigating and it looks like when you emailed us a few weeks ago you sent an email directly to shipping at cohesion3d dot com. That is an unmonitored email address just used for the shipping system to send the shipment notification emails from. I have a reply-to set such that if you reply to that email, it'll go to our actual helpdesk which is info at cohesion3d dot com. But since you sent an email directly to shipping, that's why I never saw your email. I'm going to change that setup, by the way, to avoid this happening ever again. And again, sorry we missed it. 

Now, your package has not moved in a while. Did you ever try to contact the local post office with the tracking number to see if maybe it was there and they were holding it for customs payment? 

I have submitted an international inquiry with USPS to look into this, and hopefully claim the partial insurance value of the package if it is indeed lost. I don't know exactly how long it will take for them to get back to me, the form said up to 32 days. If you can shoot me an email to info at cohesion3d dot com I can keep you up to date on any progress updates with this and we can discuss further - I would like to give the postal system at least some time to look into this before we take next steps such as getting a new order sent out to you.

Priority Mail to the UK has been mostly solid so far, I think this is the first such problem we are dealing with out of hundreds shipped to the UK via USPS. 

Looking forward to your thoughts.

Many thanks,
Ray Kholodovsky




My reply:

Hi Ray,

Thanks for your prompt response to our PayPal dispute.

I have contacted Parcelforce, who seem to be USPS' main carrier in the UK and who do have the traacking number logged with them. However, the package has never been received by them and is now beyond the time period where it is their issue anymore, unfortunately.

With regards to contacting the local post office, I'm afraid we wouldn't begin to have a clue who that would be, and have had no contact whatsoever regarding customs requirements or anything like that. It's proven impossible to contact USPS dorectly as the only contact forms they have require a US Zip and State for delivery, and obviously we don't have those. An email sent to them directly was responded to by telling us to fill in that form, so I can only assume it was automated.

We're happy to do what we can to get the parcel delivered and avoid any unnecessary costs on either side, so if there is a way to work together to do that, that would be great.

Thanks again for your email, and I look forward to getting this issue ironed out.

Many Thanks,

Seán Gleeson




Hopefully we're on our way to getting this figured out.

**Comment by DoESsean** [comment]
More emails!


_Hi Sean,

Thanks for looking into things and providing that information.  Other customers have also mentioned Parcelforce as the entity that handles the delivery of USPS priority packages, so it sounds like you went down the correct road. 

As I mentioned via the PayPal form, I submitted a USPS International Inquiry.  USPS Priority comes with $200 of insurance, so I'm hoping to get that amount back from them.  The inquiry process says it can take 32 days, and I've never had to deal with this before, so I don't know how that's going to go or how much longer after they are going to take to pay out the claim.
​
I'm not going to ask you to wait another full month.  I'd like to give it a few weeks to see if they come up with anything or the package magically starts moving.  The reality of the situation is that I'm most likely going to eat the cost of all this and send you guys a new set of what you ordered.  

Is this ok?

Cheers,
Ray Kholodovsky
Cohesion3D_


And my reply...

_Hi Ray,

Thanks for the reply

That would be fantastic, thank you very much. Obviously I'd rather this didn't cost anyone anything more than it should, and hopefully you'll be able to get the package returned to you to resell, but it will be good to get our laser up and moving.

If you can keep me updated with any response from USPS so we know what it happening, that would be great - it may be that you have more luck getting a useful answer from them than I can from this side of the Atlantic, and we can simply get the original package moving in the right direction.

Thanks again,

Seán_


Basically, we're moving forward and should have this sorted relatively soon.

**Comment by DoESsean** [comment]
Exciting update:

The package has been resent. Currently, the tracking is showing it as having departed Heathrow and being on its way to the destination, so I'll keep an eye on it and if it stops moving for a couple of days, get straight in touch.

Thanks for your patience, hopefully this will be sorted soon.

**Comment by DoESsean** [comment]
Even more exciting update:

The package is being held at customs in Coventry while they do the checks and stuff, so we'll hopefully find out soon what needs paying, and then get it delivered. This is a whole extra step from last time, so hopefully it means it's actually happening this time around.

**Comment by magman2112** [comment]
Post office customs charge letter has arrived. Jackie opened it and it was less than £50, which was a bit of a bonus. 

Parcel is in Huyton Depot, so once custom charges are paid it should finally be in our hands in a few days.

**Comment by Sean-anotherone** [comment]
Woohoo!  most excellent news 

Some alarm cable, or similar thin multicore, will also be required as a few interlocks will need installing.
about 15 m would probably be plenty for 2 x microswitches (I've provided those already) and the main laser-enable interlock.

Other things to find :

a switch for the led cabinet lighting
A SSD or relay for the blow compressor.
a length of flex duct for the extract fan (perhaps, might have enough for a single-point extract)

**Comment by DoESsean** [comment]
Customs Charges have been paid, delivery is scheduled for tomorrow (Wednesday 14th).

**Comment by ajlennon** [comment]
omg the excitement!

**Comment by amcewen** [comment]
![Unopened parcel from the US](https://pbs.twimg.com/media/EB8BjwlWsAEe6yK?format=jpg&name=4096x4096)

**Comment by Sean-anotherone** [comment]
da da da duuuuummmmmmmmmmm

**Comment by magman2112** [comment]
The controller is safely here.

@Sean-anotherone and I have done some initial planning for the fitting of parts and the following tasks are added to the stack for this project.

- [ ] Modify the existing 24V AC power supply to provide 24V DC to power the Servo motors. Mike to provide Panel mount Electrolytic capacitors and a bridge rectifier to support this.
- [ ] Move Servo drivers and rewire connections to Servo motors as required - to make it easier to connect to the  Laser controller
- [ ] Mike to make a mounting plate for the Laser controller, with appropriate standoffs for the PCB.
- [ ] Install relay and mains power connections to enable the Laser Controller to control the Air Assist.
- [ ] Install mains power from Circuit breakers for Laser Controller PSU.
- [ ] Obtain and install new silicone tubing for the air assist.
- [ ] Look at a better installation for the LED illumination and power supply, replacing the power supply if appropriate.

Additional tasks will be added as they are identified.

**Comment by amcewen** [comment]
Recently @Sean-anotherone mentioned it might be useful to have some multicore cable for the wiring.  Once I get back from London I'll have about half a reel of [this 15-core 20AWG cable](https://uk.farnell.com/alpha-wire/1896-15c-sl005/cable-20awg-15-core-slate-30-5m/dp/1744095) available if it's any use?

**Comment by magman2112** [comment]
Closing this issue now as the electronics and software are now working on this laser.

Commissioning issues are being addressed in #955

**Comment by amcewen** [comment]
:tada: Well done @magman2112, @Sean-anotherone and everyone else who worked on getting this operational!

**Comment by ajlennon** [comment]
Yeah to echo that well done folks. You are all awesome x
